arch/rv32i/src/syscall.rs

```diff
@@ -40,6 +42,35 @@ const R_A2: usize = 11;
 const R_A3: usize = 12;
 const R_A4: usize = 13;
 
+const MAJOR_VER: usize = 1;
+const MINOR_VER: usize = size_of::<Riscv32iStoredState>();
+const TAG: [u8; 4] = [b'r', b'v', b'3', b'2'];
```

Comment:
I like the idea of using meaningful semantic information in the tag, though we're going to want to avoid some obvious collisions if we can :smile:. Given we have 4 bytes here, and the state to store heavily depends on the extensions present on RISC-V (e.g. floating point registers), maybe it'd be a good idea to use something like `r5i\0`, where `r` -> RISC-V, `5` -> 2^5 bits, `i\0` for the base integer instruction set.